### PR TITLE
MESH-1759 Remove porfolio name to number mapping after deletion

### PR DESCRIPTION
**Pull request by raycar5**
### modified logic

The portfolio name is now removed from the NameToNumber mapping in the `delete_portfolio` extrinsic.

### CODE REVIEW

**Review by Centril — APPROVED** [review]

**Review by CJP10 — APPROVED** [review]

**Review by Neopallium — CHANGES_REQUESTED** [review]
This should have a storage migration to cleanup deleted portfolio.

File: pallets/portfolio/src/lib.rs [hunk 1]
```diff
@@ -330,6 +336,32 @@ decl_module! {
...
+                    if Portfolios::get(identity,number) != name{
+                        Some((identity,name))
+                    }else{
+                        None
+                    }
+                });
+
+                for (identity,name) in must_remove {
+                    NameToNumber::remove(identity,name)
+                }
```
Comment (Centril) [inline pallets/portfolio/src/lib.rs:360]:
```suggestion
                NameToNumber::iter()
                    .filter(|(identity, name, number)| !Portfolios::contains_key(identity, number))
                    .for_each(|(identity, name, _)| NameToNumber::remove(identity, name));
```

File: pallets/portfolio/src/lib.rs [hunk 2]
```diff
@@ -330,6 +336,32 @@ decl_module! {
         pub fn accept_portfolio_custody(origin, auth_id: u64) -> DispatchResult {
             Self::base_accept_portfolio_custody(origin, auth_id)
         }
+
+        fn on_runtime_upgrade() -> Weight {
+            use polymesh_primitives::{ storage_migrate_on };
```
Comment (Centril) [inline pallets/portfolio/src/lib.rs:341]:
```suggestion
            use polymesh_primitives::storage_migrate_on;
```

File: pallets/portfolio/src/lib.rs [hunk 3]
```diff
@@ -191,7 +195,9 @@ decl_module! {
             Self::ensure_portfolio_custody_and_permission(pid, primary_did, secondary_key.as_ref())?;
 
             // Delete from storage.
+            let portfolio =Portfolios::get(&primary_did,&num);
```
Comment (Centril) [inline pallets/portfolio/src/lib.rs:198]:
```suggestion
            let portfolio = Portfolios::get(&primary_did, &num);
```

**Review by Centril — APPROVED** [review]
Logic looks good, this just simplifies + formats code a bit.

File: pallets/portfolio/src/lib.rs [hunk 4]
```diff
@@ -338,26 +338,17 @@ decl_module! {
...
-                    }else{
-                        None
-                    }
-                });
-
-                for (identity,name) in must_remove {
-                    NameToNumber::remove(identity,name)
-                }
+                NameToNumber::iter()
+                    .filter(|(identity, name, number)| !Portfolios::contains_key(identity, number))
```
Comment (Centril) [inline pallets/portfolio/src/lib.rs:350]:
```suggestion
                    .filter(|(identity, _, number)| !Portfolios::contains_key(identity, number))
```

**Review by Neopallium — APPROVED** [review]

**Pull request merged**